service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadAdditionalApplicationsController.java

```diff
@@ -0,0 +1,155 @@
+import uk.gov.hmcts.reform.fpl.model.FeesData;
+import uk.gov.hmcts.reform.fpl.model.common.C2DocumentBundle;
+import uk.gov.hmcts.reform.fpl.service.PbaNumberService;
+import uk.gov.hmcts.reform.fpl.service.UploadC2DocumentsService;
+import uk.gov.hmcts.reform.fpl.service.payment.FeeService;
+import uk.gov.hmcts.reform.fpl.service.payment.PaymentService;
+import uk.gov.hmcts.reform.fpl.utils.BigDecimalHelper;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import static uk.gov.hmcts.reform.fpl.enums.ApplicationType.C2_APPLICATION;
+import static uk.gov.hmcts.reform.fpl.enums.YesNo.NO;
+import static uk.gov.hmcts.reform.fpl.enums.YesNo.YES;
+
+@Api
+@Slf4j
+@RestController
+@RequestMapping("/callback/upload-additional-applications")
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class UploadAdditionalApplicationsController extends CallbackController {
+    private static final String DISPLAY_AMOUNT_TO_PAY = "displayAmountToPay";
+    private static final String AMOUNT_TO_PAY = "amountToPay";
+    private static final String TEMPORARY_C2_DOCUMENT = "temporaryC2Document";
+    private final ObjectMapper mapper;
+    private final FeeService feeService;
+    private final PaymentService paymentService;
+    private final PbaNumberService pbaNumberService;
+    private final UploadC2DocumentsService uploadC2DocumentsService;
+
+    @PostMapping("/get-fee/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEvent(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        Map<String, Object> data = caseDetails.getData();
+        CaseData caseData = getCaseData(caseDetails);
+        data.remove(DISPLAY_AMOUNT_TO_PAY);
+
+        //workaround for previous-continue bug
+        if (shouldRemoveDocument(caseData)) {
+            removeDocumentFromData(data);
+        }
+
+        try {
+            FeesData feesData = feeService.getFeesDataForC2(caseData.getC2ApplicationType().get("type"));
+            data.put(AMOUNT_TO_PAY, BigDecimalHelper.toCCDMoneyGBP(feesData.getTotalAmount()));
+            data.put(DISPLAY_AMOUNT_TO_PAY, YES.getValue());
+        } catch (FeeRegisterException ignore) {
+            data.put(DISPLAY_AMOUNT_TO_PAY, NO.getValue());
+        }
+
+        return respond(caseDetails);
+    }
+
+    @PostMapping("/validate/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleValidateMidEvent(
+        @RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = getCaseData(caseDetails);
+
+        var updatedPbaNumber = pbaNumberService.update(caseData.getPbaNumber());
+        caseDetails.getData().put("pbaNumber", updatedPbaNumber);
+        List<String> errors = new ArrayList<>();
+        errors.addAll(pbaNumberService.validate(updatedPbaNumber));
```

Comment:
can be just 
 List<String> errors = pbaNumberService.validate(updatedPbaNumber);